### PR TITLE
feat: Create Select's foundational component - SelectValue

### PR DESCRIPTION
**Pull request by brianacnguyen**
## **Description**
- Added SelectValue - foundational component for Select

## **Related issues**
Part of #6809 and #7242 

## **Screenshots/Recordings**
https://github.com/MetaMask/metamask-mobile/assets/14355083/5945ba4f-69ad-44aa-b2f5-ccd9cceacb2f

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by georgewrmarshall — DISMISSED** [review]
LGTM! Nice work. Left some non-blocking comments that can be resolved in future it also seems that SelectValueBase is missing a README file
- Pulled branch and ran storybook locally

**Comment by georgewrmarshall** [comment]
Can you also link the issue? Maybe "Part of: #" would be appropriate if there is no direct issue for this component

File: app/component-library/components/Select/SelectValue/SelectValue.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import { default as SelectValueComponent } from './SelectValue';
+import { SAMPLE_SELECTVALUE_PROPS } from './SelectValue.constants';
+
+const SelectValueStoryMeta = {
+  title: 'Component Library / Select',
+  component: SelectValueComponent,
+  argTypes: {
+    label: {
+      control: { type: 'text' },
+      defaultValue: SAMPLE_SELECTVALUE_PROPS.label,
```
[the remark this answers is not in the document]
Comment (brianacnguyen) [reply app/component-library/components/Select/SelectValue/SelectValue.stories.tsx:15]: once its deprecated ill do a sweep to update all the existing stories to not use it

File: app/component-library/components/Select/SelectValue/README.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+Optional prop for the start Icon Element.
+
+| <span style="color:gray;font-size:14px">TYPE</span> | <span style="color:gray;font-size:14px">REQUIRED</span> |
+| :-------------------------------------------------- | :------------------------------------------------------ |
+| Avatar-like Elements                                              | No                                                     |
+
+### `iconProps`
+
+Optional prop for the start Icon Element.
```
[the remark this answers is not in the document]
Comment (brianacnguyen) [reply app/component-library/components/Select/SelectValue/README.md:19]: Optional is the mobile standards for optional props

File: app/component-library/components/Select/SelectValue/SelectValue.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+}) => {
+  const { styles } = useStyles(styleSheet, {
+    style,
+  });
+
+  const renderLabel = () =>
+    typeof label === 'string' ? (
+      <Text
+        variant={DEFAULT_SELECTVALUE_LABEL_TEXTVARIANT}
+        color={DEFAULT_SELECTVALUE_LABEL_TEXTCOLOR}
```
[the remark this answers is not in the document]
Comment (brianacnguyen) [reply app/component-library/components/Select/SelectValue/SelectValue.tsx:41]: that pattern is currently not followed by mobile so im not gonna add it here

**Review by georgewrmarshall — APPROVED** [review]

**Pull request merged**